CreditManagement/views.py

```diff
@@ -15,44 +16,201 @@ class TransactionListView(ListView):
+
+            return HttpResponseRedirect(request.GET.get('redirect', request.path_info))
+
+        context = self.get_context_data()
+        context['form'] = form
+        return render(request, self.template_name, context)
+
+
+class TransactionUserView(TransactionAlterView):
+    def get_form(self, transaction=None, data=None):
+        return UserTransactionForm(self.request.user,
+                                   initial_from_get=self.request.GET,
+                                   data=data,
+                                   instance=transaction)
+
+    def check_access_permission(self, request):
+        transaction_id = request.GET.get('id', None)
+
+        if transaction_id is not None:
+            # get the instance
+            t_order = get_object_or_404(PendingTransaction, pk=transaction_id)
+            if t_order.source_user != request.user:
+                return HttpResponseForbidden("You do not have access to this transaction")
+            if t_order.confirm_moment <= timezone.now():
+                return HttpResponseForbidden("This transaction can no longer be altered")
+
+        return super(TransactionUserView, self).check_access_permission(request)
+
+
+class DonationView(TransactionUserView):
+    def get_form(self, transaction=None, data=None):
+        if transaction is None:
+            initial = {'amount': 0.5}
+        else:
+            initial = {}
+
+        return UserDonationForm(self.request.user,
+                                initial_from_get=self.request.GET,
+                                initial=initial,
+                                data=data,
+                                instance=transaction)
+
+
+class TransactionAssociationView(TransactionAlterView):
+    def check_access_permission(self, request):
+        association_name = self.kwargs.get('association_name')
+        association = get_object_or_404(Association, slug=association_name)
+
+        transaction_id = request.GET.get('id', None)
+
+        if transaction_id is not None:
             # If an association is given as the source, check user credentials
-            if not request.user.is_board_of(association.id):
-                return HttpResponseForbidden()
-            # Create the form
-            form = AssociationTransactionForm(association, request.POST)
+            if not self.request.user.is_board_of(association.id):
+                return HttpResponseForbidden("You do not have access to this transaction")
+            t_order = get_object_or_404(PendingTransaction, pk=transaction_id)
+            if t_order.confirm_moment <= timezone.now():
+                return HttpResponseForbidden("This transaction can no longer be altered")
         else:
-            form = UserTransactionForm(request.user, request.POST)
+            if not self.request.user.is_board_of(association.id):
+                return HttpResponseForbidden("You can not make transactions for this association")
```

Comment:
> But you know, when someone starts looking for leaks...
???

This kind of insinuates that you think that I suggested that HttpResponseForbidden is not really necessary just because it's not possible to get there by following links. Do you really think that that's what I meant?...

Always return HttpResponseForbidden, but the message provided is only for programming debugging reasons and (as I explained above) is not displayed to the user, nor should it be. We should not be wasting code on guiding people who are not using the UI.